### PR TITLE
Embedded object constraints

### PR DESCRIPTION
**Pull request by tanaya-mankad**
This branch has already merged the 232-updates branch.
In addition, it includes 3 separate changes:

1. the schema translation Python uses pathlib instead of os.path,
2. exceptions in translation pieces propagate properly to the interface,
3. source schema are first parsed into a new object-type-keyed data structure to make JSON translation flow better.

### CODE REVIEW

File: examples/fan_spec/schema/RS0006.schema.yaml [hunk 1]
```diff
@@ -82,22 +82,22 @@ Performance:
       Required: True
     performance_map:
       Description: "Data group describing drive performance when operating"
-      Data Type: "{PerformanceMap}"
+      Data Type: "{PerformanceMapDrive}"
```
Comment (nealkruis) [inline examples/fan_spec/schema/RS0006.schema.yaml:85]: Looks like you missed some...

File: examples/fan_spec/schema/RS0006.schema.yaml [hunk 2]
```diff
@@ -82,22 +82,22 @@ Performance:
...
       Required: True
 
 PerformanceMapDrive:
```
Comment (nealkruis) [inline examples/fan_spec/schema/RS0006.schema.yaml:88]: ☝🏻

File: examples/fan_spec/schema/RS0007.schema.yaml [hunk 3]
```diff
@@ -76,22 +76,22 @@ Performance:
       Required: True
     performance_map:
       Description: "Data group describing drive performance when operating"
-      Data Type: "{PerformanceMap}"
+      Data Type: "{PerformanceMapDrive}"
```
Comment (nealkruis) [inline examples/fan_spec/schema/RS0007.schema.yaml:79]: ☝🏻

File: examples/fan_spec/schema/RS0007.schema.yaml [hunk 4]
```diff
@@ -76,22 +76,22 @@ Performance:
...
       Required: True
 
-PerformanceMapMechDrive:
+PerformanceMapDrive:
```
Comment (nealkruis) [inline examples/fan_spec/schema/RS0007.schema.yaml:82]: ☝🏻

File: examples/lookup_table/schema/LookupTable.schema.yaml [hunk 5]
```diff
@@ -3,9 +3,17 @@ Schema:
...
-LookupTable:
+SchemaType:
+  Object Type: "Enumeration"
+  Enumerators:
+    TYPE1:
+      Description: "Type 1"
+    TYPE2:
+      Description: "Type 2"
+
+LookupTableTemplate:
```
Comment (nealkruis) [inline examples/lookup_table/schema/LookupTable.schema.yaml:16]: ❤️

**Review by nealkruis — CHANGES_REQUESTED** [review]

**Comment by tanaya-bigladder** [comment]
Oops, fixed now.

**Comment by nealkruis** [comment]
@tanaya-mankad, is this PR now obsolete?

**Comment by tanaya-mankad** [comment]
PR is closed; I'll make sure the branch has nothing left I want to merge and then delete it as well.

**Pull request closed without merging**